### PR TITLE
perf: Reduce chain calls of platform-express

### PR DESCRIPTION
**Pull request by LaicZhang**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [12081](https://github.com/nestjs/nest/issues/12081)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

**Comment by micalevisk** [comment]
I'd love to see some benchmarks on this because the changes on `.interceptor.ts` files seems to be micro-optimisation to me

File: packages/platform-express/adapters/express-adapter.ts [hunk 1]
```diff
@@ -63,18 +63,17 @@ export class ExpressAdapter extends AbstractHttpAdapter<
...
+    const logger = this.logger;
     if (statusCode) {
       response.status(statusCode);
     }
     if (isNil(body)) {
       return response.send();
     }
+    const contentType = response.getHeader('Content-Type');
     if (body instanceof StreamableFile) {
       const streamHeaders = body.getHeaders();
```
Comment (vlourenco-stf) [inline packages/platform-express/adapters/express-adapter.ts:75]: I think you can replace this with `.getHeader()` for each header as calling them one by one 4 times will be faster than having all the headers.

From what I remember, `.getHeaders` clones all headers, so it's a waste of memory if you only want 4 headers.

**Comment by vlourenco-stf** [comment]
I agree with @micalevisk, except reusing the content type and reducing `.getHeader` calls, the other optimizations are more cleanup than real optimization (unless you have some link/reference for those kind of optimizations, I'd love to know more).

The `switch` case is another thing that looks more like a cleanup, a valid change as it looks a lot cleaner, but I don't think it changes performance in any relevant way. The only exception is when the value in switch case is a typescript enum, so this change can optimize the code, enums are really bad when you import them from a file with `export *`.

File: packages/platform-express/multer/interceptors/file-fields.interceptor.ts [hunk 2]
```diff
@@ -64,6 +64,6 @@ export function FileFieldsInterceptor(
       return next.handle();
     }
   }
-  const Interceptor = mixin(MixinInterceptor);
-  return Interceptor;
+
+  return mixin(MixinInterceptor);
```
Comment (kamilmysliwiec) [inline packages/platform-express/multer/interceptors/file-fields.interceptor.ts:68]: I don't think this has any impact on the performance (code interpreter should auto-optimize this either way)

File: packages/platform-express/multer/interceptors/file.interceptor.ts [hunk 3]
```diff
@@ -62,6 +62,6 @@ export function FileInterceptor(
       return next.handle();
     }
   }
-  const Interceptor = mixin(MixinInterceptor);
-  return Interceptor;
+
+  return mixin(MixinInterceptor);
```
Comment (kamilmysliwiec) [inline packages/platform-express/multer/interceptors/file.interceptor.ts:66]: https://github.com/nestjs/nest/pull/12094/files#r1271850425

File: packages/platform-express/multer/interceptors/files.interceptor.ts [hunk 4]
```diff
@@ -65,6 +65,6 @@ export function FilesInterceptor(
       return next.handle();
     }
   }
-  const Interceptor = mixin(MixinInterceptor);
-  return Interceptor;
+
+  return mixin(MixinInterceptor);
```
Comment (kamilmysliwiec) [inline packages/platform-express/multer/interceptors/files.interceptor.ts:69]: https://github.com/nestjs/nest/pull/12094/files#r1271850425

File: packages/platform-express/multer/interceptors/no-files.interceptor.ts [hunk 5]
```diff
@@ -56,6 +56,6 @@ export function NoFilesInterceptor(
       return next.handle();
     }
   }
-  const Interceptor = mixin(MixinInterceptor);
-  return Interceptor;
+
+  return mixin(MixinInterceptor);
```
Comment (kamilmysliwiec) [inline packages/platform-express/multer/interceptors/no-files.interceptor.ts:60]: https://github.com/nestjs/nest/pull/12094/files#r1271850425

**Pull request closed without merging**